src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/Order2DeliveryForm.tsx

```diff
@@ -96,9 +100,27 @@ export const Order2DeliveryForm: React.FC<Order2DeliveryFormProps> = ({
       formikHelpers: FormikHelpers<FormikContextWithAddress>,
     ) => {
       try {
+        setCheckoutMode("standard")
         checkoutTracking.clickedOrderProgression(
           ContextModule.ordersFulfillment,
         )
+
+        if (orderData.selectedFulfillmentOption?.type) {
+          // Unset the current fulfillment option if it exists
+          const unsetFulfillmentOptionResult =
+            await unsetOrderFulfillmentOption.submitMutation({
+              variables: {
+                input: {
+                  id: orderData.internalID,
+                },
+              },
+            })
+          validateAndExtractOrderResponse(
+            unsetFulfillmentOptionResult.unsetOrderFulfillmentOption
+              ?.orderOrError,
+          )
+        }
```

Comment:
Fix to making order totals and fulfillment option reset if it was already on pickup (or ship for a different address). This is kind of tricky because exchange has its own logic about setting and recalculating things when setting a shipping address but that logic is primarily concerned with server-side changes rather than client needs.